src/client.ts

```diff
@@ -2595,6 +2599,85 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     );
   }
 
+  async queryThreads(options?: {
+    limit?: number;
+    next?: string;
+    participant_limit?: number;
+    prev?: string;
+    reply_limit?: number;
+    watch?: boolean;
+  }) {
+    // eslint-disable-next-line
+    const opts = {
+      limit: 10,
+      participant_limit: 10,
+      reply_limit: 3,
+      watch: true,
+      ...options,
+    };
+
+    const res = await this.post<QueryThreadsAPIResponse<StreamChatGenerics>>(this.baseURL + `/threads`, opts);
+    const threads: Thread<StreamChatGenerics>[] = [];
```

Comment:
Could the following lines be simplified a bit?
```suggestion
    return {
      threads: res.threads.map((thread) => new Thread(this. thread)),
      next: res.next,
    };
```